### DISCUSSION TITLE
How to handle user data in Supabase when authentication is done in Firebase?

### DISCUSSION THREAD

**Discussion by pradeepb28** [supabase/supabase#10024] [Questions] [answered]
hello everyone,

tldr;
Due to lack of APIs or not up to the mark to built backend for iOS app, I went with Firebase for authentication and user generated data in Supabase.
1. Firebase for authentication
2. Supabase for user data

I am in a tricky position right now in handling user data as I am still a beginner in backend development.

- How would you ensure that data is saved on db in supabase as well as handling API requests to ensure user is seeing their own data but not from others. 
- What authentication or required data needs to be sent from Firebase to Supabase so that signup, signin and user data is shown accordingly.

Any help is highly appreciated 🙏

**Comment by hf** [comment, marked as answer]
Hey @pradeepb28,

This is technically possible but does require quite a bit of infrastructure on your side (a translation layer to translate Firebase JWTs into Supabase JWTs). We don't recommend doing this. 

It's not possible to use the Firebase JWTs as a replacement for those issued by Supabase Auth at this time.

You can import Firebase users into Supabase Auth, if you want.

**Reply by pradeepb28** [reply]
>You can import Firebase users into Supabase Auth, if you want.

Thanks for the quick reply. I wish I could that but like I said the current Supabase SDK for iOS is not up to minimum standards so for now we can't pick the SDK. I thought of going with non SDK approach but I couldn't find any documentation on website on how to handle supabase auth either by manual email or social signups.

**Comment by sunilguptasg** [comment]
Hi,

I used Firebase auth, and used a cloud function to add the Supabase JWT to the returned auth (by setting the JWT using setCustomUserClaims). This function is called only when a new user logs in (saving billable function calls). 

This then works by setting Supabase.instance.client.auth.setAuth(token).

Issue - cannot figure out how to do this in 1.x - setAuth has been removed.
